syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -423,11 +443,19 @@ def __itruediv__(self, *args, **kwargs):
 
     @overloaded.method
     def __truediv__(self, shares: dict, divisor):
+        # TODO: how to correctly handle division in Zq?
         assert isinstance(divisor, int)
 
         divided_shares = {}
-        for location, pointer in shares.items():
-            divided_shares[location] = pointer / divisor
+        for i_worker, (location, pointer) in enumerate(shares.items()):
+            # Still no solution to perform a real division on a additive shared tensor
+            # without a heavy crypto protocol.
+            # For now, the solution works in most cases (we compute Q - (Q - pointer) / divisor
+            # for as many worker as the number of times the sum of shares "crosses" Q/2, in average)
+            if i_worker % 3 == 0:
```

Comment:
Why 3 here ? The theorem 1 of https://eprint.iacr.org/2017/396.pdf is for 2 parties, have you tried with more?

Reply:
It's a bit hard to explain and I'm not 100% sure it's true but I think that for more shares, one third of them should be transformed with `Q - (Q - val) / div`. It's because it's needed as many times as the sum of shares crosses `Q / 2` when wrapped around the field. As values are concentrated around 0 for a big field and shares are generated randomly, I think that the number of times `Q / 2` is crossed is indeed one third of the number of shares.
Tell me if this is not clear, or even not true.